lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/StorageClient.java

```diff
@@ -48,28 +48,28 @@ public interface StorageClient {
    * Gets the appropriate client based on the URI scheme and validate with authentication/settings from cloudOptions
    */
   static StorageClient create(URI pathToStorage, String docIdPrefix, List<Pattern> excludes, List<Pattern> includes,
-      Map<String, Object> cloudOptions, Config fileOptions) {
+      Map<String, Object> cloudOptions, Config fileOptions, Publisher publisher) {
```

Comment:
Is this change necessary for resolving the performance issue that this PR is addressing?
There was an explicit decision during the design & review of the FileConnector framework that StorageClient should not bound to any one Publisher. A StorageClient knows about a storage backend and lets you access files from that backend. Once you have a StorageClient you can tell it to publish files with a given publisher but you shouldn't need a publisher in order to create a StorageClient. If there's a concern here, can we address this in a separate ticket/PR?

Reply:
Here is a way to implement the essential changes in this PR without adding Publisher everywhere
https://github.com/kmwtechnology/lucille/pull/244